corehq/apps/users/views/__init__.py

```diff
@@ -1127,22 +1127,34 @@ def page_url(self):
         return reverse(self.urlname, args=self.args, kwargs=self.kwargs)
 
 
-@require_can_edit_web_users
-def web_user_upload_job_poll(request, domain, download_id,
-                             template="users/mobile/partials/user_upload_status.html"):
-    try:
-        context = get_download_context(download_id)
-    except TaskFailedError:
-        return HttpResponseServerError()
-
-    context.update({
-        'on_complete_short': _('Bulk upload complete.'),
-        'on_complete_long': _('Web Worker upload has finished'),
-        'user_type': _('web users'),
-    })
+class UserUploadJobPollView(BaseManageWebUserView):
 
-    context['result'] = BulkUploadResponseWrapper(context)
-    return render(request, template, context)
+    def get(self, request, domain, download_id):
+        try:
+            context = get_download_context(download_id)
+        except TaskFailedError:
+            return HttpResponseServerError()
+
+        context.update({
+            'on_complete_short': _('Bulk upload complete.'),
+            'on_complete_long': _(self.on_complete_long),
+            'user_type': _(self.user_type),
+        })
+
+        context['result'] = BulkUploadResponseWrapper(context)
+        return render(request, 'users/mobile/partials/user_upload_status.html', context)
+
+
+class WebUserUploadJobPollView(UserUploadJobPollView):
+    urlname = "web_user_upload_job_poll"
+
+    def __init__(self):
```

Comment:
these can be directly on the class (like `urlname`) and then won't have to worry about an `init` method, which might mess with the existing init. If it does use one, you should call the super class init, to make sure this doesn't wipe out any existing setup

Reply:
[1ab2193](https://github.com/dimagi/commcare-hq/pull/28832/commits/1ab2193aae3bbb17331ccb61f7fa77c3ac619e85) great point-- added directly to the class